### PR TITLE
SSB: Prevent salamander's ability from crashing in edge cases

### PR DESCRIPTION
**Pull request by Salaaa**

### CODE REVIEW

File: data/mods/ssb/abilities.js [hunk 1]
```diff
@@ -998,7 +998,7 @@ let BattleAbilities = {
 		desc: "This Pokemon is immune to volatile statuses.",
 		shortDesc: "This Pokemon is immune to volatile statuses.",
 		onTryAddVolatile(status, target) {
-			if (toID(target.name).includes(status.id)) return;
+			if ([toID(target.name), 'furycutter', 'stockpile'].includes(status.id)) return;
```
Comment (Zarel) [inline data/mods/ssb/abilities.js]: Why isn't `toID(target.name)` hardcoded? It's just `'salamander'`, right?

Comment (Salaaa) [reply data/mods/ssb/abilities.js]: Fair enough

Comment (HoeenCoder) [reply data/mods/ssb/abilities.js]: It was not hardcoded to handle any case where this ability could be applied to another pokemon. I don't think anything could cause that now but it prevents bugs in the case something like skill swap is introduced in the future.

Comment (Salaaa) [reply data/mods/ssb/abilities.js]: the entire purpose of this pull request was to handle a bug from a skill swap, but like, if you want I can just revert the previous commit

Comment (HoeenCoder) [reply data/mods/ssb/abilities.js]: Oh right we have metronome-like move(s). Yeah this would be needed.

**Pull request merged**